hibernate-core/src/main/java/org/hibernate/metamodel/mapping/internal/DiscriminatedAssociationAttributeMapping.java

```diff
@@ -75,9 +73,8 @@ public DiscriminatedAssociationAttributeMapping(
 				bootProperty.getName(),
 				stateArrayPosition,
 				attributeMetadataAccess,
-				fetchTiming == FetchTiming.IMMEDIATE
-						? new FetchStrategy( FetchTiming.IMMEDIATE, FetchStyle.SELECT )
-						: new FetchStrategy( FetchTiming.DELAYED, FetchStyle.SELECT ),
+				fetchTiming,
+				FetchStyle.SELECT,
 				declaringType,
```

Comment:
The above is a perfect example why this PR is necessary.

Reply:
Yes, cases like this are exactly why I created this task